File: 프로그래머스/Level2/숫자의 표현.py
# https://programmers.co.kr/learn/courses/30/lessons/12924

def solution(n):
    answer = 1
    numbers = [i for i in range(1, n)]
    for i in range(len(numbers) - 1):
        for j in range(i+1, len(numbers)):
            temp_sum = sum(numbers[i : j])
            if n == temp_sum:
                answer += 1
                break
            if n < temp_sum:
                break
    return answer

File: 프로그래머스/Level2/숫자의 표현.js
// https://programmers.co.kr/learn/courses/30/lessons/12924

function solution(n) {
	let answer = 1;
	for (let i = 1; i < n; i++) {
		let temp = i;
		for (let j = i + 1; j < n; j++) {
			temp += j;
			if (temp === n) {
				answer++;
				break;
			} else if (temp > n) break;
		}
	}
	return answer;
}
